community/values/src/main/java/org/neo4j/values/storable/PointValue.java

```diff
@@ -168,9 +169,9 @@ else if ( this.coordinate.length < other.coordinate.length )
     }
 
     @Override
-    public Object asObjectCopy()
+    public Point asObjectCopy()
     {
-        return new PointValue( this.crs, this.coordinate );
+        return this;
```

Comment:
I am unsure if it is OK not to create a new object (as done with `clone`) in other implementations of `asObjectCopy`.

Reply:
If you read the rest of the code, you will find that `PointValue` is never mutated, and that the `coordinate` field is not made public to anything that can mutate it.

Would applying this diff make you more comfortable?

```
--- a/community/values/src/main/java/org/neo4j/values/storable/PointValue.java
+++ b/community/values/src/main/java/org/neo4j/values/storable/PointValue.java
@@ -32,8 +32,8 @@ import static java.util.Collections.singletonList;
 
 public class PointValue extends ScalarValue implements Comparable<PointValue>, Point
 {
-    private CoordinateReferenceSystem crs;
-    private double[] coordinate;
+    private final CoordinateReferenceSystem crs;
+    private final double[] coordinate;
 
     PointValue( CoordinateReferenceSystem crs, double... coordinate )
     {
```

Reply:
Looking further at the usages of the `coordinate()` method through which the `double[] coordinate` field might leak, I think it would make a lot of sense to instead expose these two methods:

```
public int numberOfCoordinates()
{
    return coordinate.length;
}
public double coordinate( int i )
{
    return coordinate[i];
}
```